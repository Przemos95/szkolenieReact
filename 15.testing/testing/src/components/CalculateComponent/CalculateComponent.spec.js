// suites / tests
// describe / it

import React from 'react';
import renderer from 'react-test-renderer';
import CalculateComponent from './CalculateComponent.dev';

describe('Caluclate Component', () => {
    it('should equal snapshot', () => {
        const component = renderer.create(<CalculateComponent name={'Tekst'} />);
        expect(component.toJSON()).toMatchSnapshot();
    });
});


describe('Testing Calc Component', () => {
    beforeAll(() => {
        // console.log("before all");
    });

    beforeEach(() => {
        // console.log("before each");
    });

    afterEach(() => {
        // console.log("after each")
    })

    afterAll(() => {
        // console.log("after all")
    })

    it('should render proper year', () => {
        expect(2 + 2).toEqual(4);
    });

    // it.skip('should render proper month', () => {
    //     expect(2 + 2).toEqual(5);
    // });

    it('should render proper month', () => {
        expect(2 + 2).toEqual(4);
    });

    it('async test', () => {
        return new Promise(resolve => setTimeout(resolve, 2000));
    })


});