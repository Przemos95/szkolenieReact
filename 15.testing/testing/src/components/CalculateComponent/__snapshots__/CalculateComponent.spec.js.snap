// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Caluclate Component should equal snapshot 1`] = `
<div>
  Witaj!
  Tekst
</div>
`;
